pkg/apps/util/util.go

```diff
@@ -276,6 +338,72 @@ func CopyApiEnvVarToV1EnvVar(in []api.EnvVar) []v1.EnvVar {
 	return out
 }
 
+func CopyPodTemplateSpecToV1PodTemplateSpec(spec *api.PodTemplateSpec) *v1.PodTemplateSpec {
+	copied, err := api.Scheme.DeepCopy(spec)
+	if err != nil {
+		panic(err)
+	}
+	in := copied.(*api.PodTemplateSpec)
+	out := &v1.PodTemplateSpec{}
+	if err := v1.Convert_api_PodTemplateSpec_To_v1_PodTemplateSpec(in, out, nil); err != nil {
+		panic(err)
+	}
+	return out
+}
+
+// HasLatestPodTemplate checks if the provided replication controller has the latest
+// deployment config pod template. If not it will return false and the string diff.
+func HasLatestPodTemplate(dc *deployapi.DeploymentConfig, rc *v1.ReplicationController) (bool, string, error) {
+	configTemplate := CopyPodTemplateSpecToV1PodTemplateSpec(dc.Spec.Template)
+	configTemplate = filterTemplateAnnotationsAndLabels(configTemplate)
+	rcCopy, err := DeploymentDeepCopyV1(rc)
+	if err != nil {
+		return true, "", err
+	}
+	rcTemplate := filterTemplateAnnotationsAndLabels(rcCopy.Spec.Template)
+	if reflect.DeepEqual(configTemplate, rcTemplate) {
+		return true, "", nil
+	}
+	return false, diff.ObjectReflectDiff(configTemplate, rcTemplate), nil
+}
+
+// HasUpdatedImages indicates that deployment configuration images were updated.
+func HasUpdatedImages(dc *deployapi.DeploymentConfig, rc *v1.ReplicationController) (bool, string) {
+	rcImages := sets.NewString()
+	for _, c := range rc.Spec.Template.Spec.Containers {
+		rcImages.Insert(c.Image)
+	}
+	for _, c := range dc.Spec.Template.Spec.Containers {
+		if !rcImages.Has(c.Image) {
+			return true, c.Image
```

Comment:
@mfojtik this will return on first updated image

We can have more images to update, right? If yes, shouldn't we wait for all of them? If not this should probably be called just `HasUpdatedImage` without `s`

Reply:
@tnozicka problem with this function is that i need to return the image that was updated... if it is multiple images then the cause is a little bit tricky to construct because it will have to contain multiple images (so multiple causes?).

Reply:
i think i will prefer multiple causes in that case, that is why causes is a list ;-)